shared/src/k8s/job.rs

```diff
@@ -0,0 +1,571 @@
+use super::super::akri::{instance::Instance, API_NAMESPACE};
+use super::{
+    pod::modify_pod_spec,
+    pod::{
+        AKRI_CONFIGURATION_LABEL_NAME, AKRI_INSTANCE_LABEL_NAME, APP_LABEL_ID, CONTROLLER_LABEL_ID,
+    },
+    OwnershipInfo, ERROR_CONFLICT, ERROR_NOT_FOUND,
+};
+use either::Either;
+use k8s_openapi::api::batch::v1::{Job, JobSpec};
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::{ObjectMeta, OwnerReference};
+use kube::{
+    api::{Api, DeleteParams, ListParams, ObjectList, PostParams, PropagationPolicy},
+    client::Client,
+};
+use log::{error, info, trace};
+use std::collections::BTreeMap;
+
+/// Find Kubernetes Jobs with a given label or field selector
+///
+/// Example:
+///
+/// ```no_run
+/// use akri_shared::k8s::job;
+/// use kube::client::Client;
+/// use kube::config;
+///
+/// # #[tokio::main]
+/// # async fn main() {
+/// let label_selector = Some("environment=production,app=nginx".to_string());
+/// let api_client = Client::try_default().await.unwrap();
+/// for job in job::find_jobs_with_selector(label_selector, None, api_client).await.unwrap() {
+///     println!("found job: {}", job.metadata.name.unwrap())
+/// }
+/// # }
+/// ```
+///
+/// ```no_run
+/// use akri_shared::k8s::job;
+/// use kube::client::Client;
+/// use kube::config;
+///
+/// # #[tokio::main]
+/// # async fn main() {
+/// let field_selector = Some("spec.nodeName=node-a".to_string());
+/// let api_client = Client::try_default().await.unwrap();
+/// for job in job::find_jobs_with_selector(None, field_selector, api_client).await.unwrap() {
+///     println!("found job: {}", job.metadata.name.unwrap())
+/// }
+/// # }
+/// ```
+pub async fn find_jobs_with_selector(
+    label_selector: Option<String>,
+    field_selector: Option<String>,
+    kube_client: Client,
+) -> Result<ObjectList<Job>, anyhow::Error> {
+    trace!(
+        "find_jobs_with_selector with label_selector={:?} field_selector={:?}",
+        &label_selector,
+        &field_selector
+    );
+    let jobs: Api<Job> = Api::all(kube_client);
+    let job_list_params = ListParams {
+        label_selector,
+        field_selector,
+        ..Default::default()
+    };
+    let result = jobs.list(&job_list_params).await;
+    trace!("find_jobs_with_selector return");
+    Ok(result?)
+}
+
+/// Create Kubernetes Job with given Instance and OwnershipInfo
+///
+/// Example:
+///
+/// ```no_run
+/// use akri_shared::k8s::{
+///     OwnershipInfo,
+///     OwnershipType,
+///     job
+/// };
+/// use akri_shared::akri::instance::{Instance, InstanceSpec};
```

Comment:
Is the sample code in the comment left on purpose?

Reply:
Yes, I provided examples for all functions in jobs.rs just like we do with pods.rs and instance.rs, etc